apps/api/src/test/java/dk/treecreate/api/exceptionhandling/ExceptionController.java

```diff
@@ -0,0 +1,35 @@
+/*
+    A helper class used by the exception handler testing for the purposes of mocking
+    Throws various exceptions based of the input
+ */
+
+package dk.treecreate.api.exceptionhandling;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+
+@RestController
+public class ExceptionController
+{
+  @PostMapping("/exception")
+  public void getSpecificException(@RequestParam(name = "exceptionName") String exceptionName,
+                                   @RequestParam(required = false, name = "customMessage") String customMessage)
+  {
+    switch (exceptionName)
+    {
+      case "ResponseStatusException":
+        if (customMessage != null)
+          throw new ResponseStatusException(HttpStatus.I_AM_A_TEAPOT, customMessage);
```

Comment:
Why are we returning "I am a Teapot" ? Wasn't this like a meme or something ?

Reply:
It is just a test. and there is nothing else in the API that would trigger it. Thus I have decided to go with it because it would for sure not interfere with anything, we unique and is a bit of a meme.